### PR TITLE
Comments' Page

### PR DESCRIPTION
**Pull request by olgapolinkina**
### Checklist
- [x] I've run `mvn test` from the root directory to see all new and existing tests pass
- [x] I've followed code style and run and ensured the code style is valid
- [x] I've created new tests if necessary.

### Motivation and Context
Comments' Page is working and ready to be moved to Production part.

Issue list: #12, #26, #28, #30, #27, #29
Tested:
* in jUnit using mockito: check work of API: getCommentsForGoalById(), setCommentForGoalById(), and also work of convertCommentToCommentDto() and test for MessageDTO. 
* in Browser: all links do what they suppose.

### Description
BACK-END:
1. Added Entity; Comment
2. Added Dto: CommentDto, MessageDto
3. Added methods to work with DB: getCommentsForGoal(), addComment()
4. Added method that send & receive from Fron-End comments: receiveCommentsForGoalById(), sendCommentForGoalById()
FRONT-END:
1. Goal.jsp page: shows already added comments and allows to add new comments.
2. Similar visual design applied to all already working pages.

### CODE REVIEW

File: src/main/java/lv/ctco/javaschool/goal/control/DtoConventer.java [hunk 1]
```diff
@@ -25,4 +27,13 @@ public static UserLoginDto convertUserToUserLoginDto(User user) {
         return dto;
     }
 
+    public static CommentDto convertCommentToCommentDto(Comment comment) {
+        CommentDto dto = new CommentDto();
+        dto.setUsername(comment.getUser().getUsername());
+        dto.setRegisteredDate(comment.getRegisteredDate().format(DateTimeConverter.formatterDateTime));
```
Comment (St-Filer) [inline src/main/java/lv/ctco/javaschool/goal/control/DtoConventer.java]: `formatterDateTime` it's field? Why without getters? It's a constant? Why doesn't follow code convention?

File: src/main/java/lv/ctco/javaschool/auth/control/UserStore.java [hunk 2]
```diff
@@ -41,6 +41,14 @@
         return user.isEmpty() ? Optional.empty() : Optional.of(user.get(0));
     }
 
+    public List<User> getUserByUsername(String user){
+        return em.createQuery(
+                "select u " +
+                        "from User u " +
+                        "where u.username like '%"+user+"%'", User.class)
```
[the remark this answers is not in the document]
Comment (olgapolinkina) [reply src/main/java/lv/ctco/javaschool/auth/control/UserStore.java]: Is this the correct way to do it?

```    public List<User> getUserByUsername(String username){
        return em.createQuery( "select u from User u " +
                        "where u.username like CONCAT('%',:username,'%')", User.class)
                .setParameter("username", username.toUpperCase())
                .getResultList();
    }

[the remark this answers is not in the document]
Comment (olgapolinkina) [reply src/main/java/lv/ctco/javaschool/auth/control/UserStore.java]: method deleted from this pull request

File: src/main/webapp/app/js/goal-page.js [hunk 3]
```diff
@@ -27,7 +27,12 @@ function onLoad() {
             'Content-Type': 'application/json'
         }
     }).then(function (response) {
-        return response.json();
+        if (response.status === 200) {
+            return response.json();
+        } else {
+            alert("Sompthing went wrong! error:404");
```
Comment (St-Filer) [inline src/main/webapp/app/js/goal-page.js]: Please fix a typo

File: src/main/webapp/app/js/goal-page.js [hunk 4]
```diff
@@ -57,8 +62,13 @@ function addComment() {
                 'Content-Type': 'application/json'
             }, body: JSON.stringify(data)
         }).then(function (response) {
-            document.getElementById("userComment").value = "";
-            onLoad();
+            if (response.status === 204) {
+                document.getElementById("userComment").value = "";
+                onLoad();
+            } else {
+                alert("Sompthing went wrong! error:404");
```
Comment (St-Filer) [inline src/main/webapp/app/js/goal-page.js]: Same typo

**Pull request merged**